### ISSUE TITLE
Sheet `to_arrow()` produces `null not null` columns

### ISSUE THREAD

**Issue by kgutwin** [ToucanToco/fastexcel#278]
With the example file [null-column.xlsx](https://github.com/user-attachments/files/16692741/null-column.xlsx):

```python
>>> excel_reader = fastexcel.read_excel('null-column.xlsx')
>>> sheet = excel_reader.load_sheet(0)
>>> sheet.to_arrow()
pyarrow.RecordBatch
record_id: double not null
nullonly: null not null
----
record_id: [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20]
nullonly: 20 nulls
```

Apparently PyArrow's [enforcement of "nullable" is not strong](https://stackoverflow.com/questions/78848723/non-nullable-field-in-schema-does-not-prevent-null-values-in-corresponding-table), so this doesn't create serious problems for me until I try to pickle and unpickle the RecordBatch. Then I get the exception:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/types.pxi", line 3532, in pyarrow.lib.field
ValueError: A null type field may not be non-nullable
```

I tried tracking down the cause of this weird `null not null` column schema, but it does appear that fastexcel [attempts to define all columns as "nullable"](https://github.com/ToucanToco/fastexcel/blob/c3cd0d9f5ca5f27e481b0775cd7826fb9fe3fee1/src/types/python/excelsheet/column_info.rs#L132), so I'm not sure where this is originating from. However, if I build the RecordBatch by way of polars, I don't get this strange schema:

```python
>>> sheet.to_polars().to_arrow().to_batches()[0]
pyarrow.RecordBatch
record_id: double
nullonly: null
----
record_id: [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20]
nullonly: 20 nulls
```

... so I suppose it's at least plausible that this could be fixed within fastexcel.

**Comment by lukapeschke** [comment]
Thank you for the clean bug report @kgutwin :pray:

**Comment by lukapeschke** [comment]
@kgutwin we just released `v0.11.6`, which includes the fix for this
